code/game/objects/items/flag.dm

```diff
@@ -23,11 +31,20 @@
 
 /obj/item/flag/update_icon()
 	overlays.Cut()
+	if(istype(src, /obj/item/flag/chameleon))
```

Comment:
yikes
`istype(src, ` is not good

Perhaps do this?
```
/obj/item/flag/proc/reset_icon()
  icon_state = initial(icon_state)

/obj/item/flag/chameleon/reset_icon()
  icon_state = C.updated_icon_state
```
And then just call `reset_icon()` in place of this `istype` block here

Reply:
Oh, that's a much better way of doing it (I couldn't think of a better way at the time XD). Will update shortly